src/Juvix/Compiler/Core/Data/BinderList.hs

```diff
@@ -12,34 +11,59 @@ data BinderList a = BinderList
 makeLenses ''BinderList
 
 fromList :: [a] -> BinderList a
-fromList l = BinderList (List.length l) (HashMap.fromList (zip [0 ..] l))
+fromList l = BinderList (length l) (HashMap.fromList (zip [0 ..] l))
 
-toList :: BinderList a -> [a]
-toList bl =
-  List.map snd $
-    sortBy (\x y -> compare (fst y) (fst x)) $
-      HashMap.toList (bl ^. blMap)
+instance Foldable BinderList where
+  foldr :: (a -> b -> b) -> b -> BinderList a -> b
+  foldr f acc bl = foldr f acc (toList bl)
 
-empty :: BinderList a
-empty = BinderList 0 mempty
+  null :: BinderList a -> Bool
+  null = (== 0) . (^. blLength)
 
-length :: BinderList a -> Int
-length = (^. blLength)
+  length :: BinderList a -> Int
+  length = (^. blLength)
+
+  toList :: BinderList a -> [a]
+  toList bl =
+    map snd $
+      sortBy (compare `on` fst) $
+        HashMap.toList (bl ^. blMap)
```

Comment:
I don't understand the extra indentation here. `null`, `length` and `toList` are not in the Foldable instance anymore?

Reply:
they are. Maybe the diff is confusing. I think it may be easier to look at the file directly

Reply:
Okay then, I thought only foldr is in the instance. Resolve the conflicts and I'll approve and merge.